libbeat/monitoring/metrics.go

```diff
@@ -12,30 +14,54 @@ type makeExpvar func() string
 // Int is a 64 bit integer variable satisfying the Var interface.
 type Int struct{ i int64 }
 
-// NewInt registers a new global integer metrics.
-func NewInt(name string) *Int {
-	return Default.NewInt(name)
+// NewInt creates and registers a new integer variable.
+//
+// Note: If the registry is configured to publish variables to expvar, the
+// variable will be available via expvars package as well, but can not be removed
+// anymore.
+func NewInt(r *Registry, name string, mode Mode) *Int {
+	if r == nil {
+		r = Default
+	}
+	v := &Int{}
+	r.Add(name, v, mode)
+	r.publish(name, makeExpvar(func() string {
+		return strconv.FormatInt(v.Get(), 10)
+	}))
+	return v
 }
 
-func (v *Int) Visit(vs Visitor) error { return vs.OnInt(v.Get()) }
 func (v *Int) Get() int64             { return atomic.LoadInt64(&v.i) }
 func (v *Int) Set(value int64)        { atomic.StoreInt64(&v.i, value) }
 func (v *Int) Add(delta int64)        { atomic.AddInt64(&v.i, delta) }
 func (v *Int) Inc()                   { atomic.AddInt64(&v.i, 1) }
 func (v *Int) Dec()                   { atomic.AddInt64(&v.i, -1) }
+func (v *Int) Visit(vs Visitor) error { return vs.OnInt(v.Get()) }
 
 // Float is a 64 bit float variable satisfying the Var interface.
 type Float struct{ f uint64 }
 
-// NewFloat registers a new global floating point metric.
-func NewFloat(name string) *Float {
-	return Default.NewFloat(name)
+// NewFloat creates and registers a new float variable.
+//
+// Note: If the registry is configured to publish variables to expvar, the
+// variable will be available via expvars package as well, but can not be removed
+// anymore.
+func NewFloat(r *Registry, name string, mode Mode) *Float {
```

Comment:
In addition to my above comment, not having `mode` removes the complexity during implementation to decide where it should belong. All metrics have the same importance, it is up to the consumer to decide which ones he wants.